### PR TITLE
Adds escape_for_javascript method

### PR DESCRIPTION
**Pull request by JustMikey**

### CODE REVIEW

File: lib/phoenix_html.ex [hunk 1]
```diff
@@ -157,4 +157,37 @@ defmodule Phoenix.HTML do
...
+
+  @escapes_for_javascript [
+    {?", "\\\""},
+    {?', "\\\'"},
+    {?\\, "\\\\"},
+    {?\n, "\\n"}   
+  ]
+
+  Enum.each @escapes_for_javascript, fn { match, insert } ->
+    defp escape_char(unquote(match)), do: unquote(insert)
```
Comment (chrismccord) [inline lib/phoenix_html.ex]: let's namespace this as `escape_js_char` or similar

File: lib/phoenix_html.ex [hunk 2]
```diff
@@ -157,4 +157,37 @@ defmodule Phoenix.HTML do
   def safe_to_string({:safe, iodata}) do
     IO.iodata_to_binary(iodata)
   end
+
+  @doc """
+  Escapes quotes (double and single) and double backslashes
+
+  This method is extremely helpful in JavaScript responses when there is a need
+  to escape html rendered from other templates, like in the following:
```
Comment (chrismccord) [inline lib/phoenix_html.ex]: "This method is useful in ..." :)

**Comment by chrismccord** [comment]
I have left just a couple small comments. Please squash your commits and I'll need @josevalim to signoff, but looks good from here. Thanks!

File: lib/phoenix_html.ex (hunk 2 above)
Comment (josevalim) [inline lib/phoenix_html.ex]: This *function*. :)

Comment (chrismccord) [inline lib/phoenix_html.ex]: 😳

**Comment by JustMikey** [comment]
Squashed the commits and added the changes

**Comment by chrismccord** [comment]
<3 <3 <3

**Comment by JustMikey** [comment]
Maybe it's worth bumping the version a little?

**Comment by josevalim** [comment]
I have other things to merge than I will bump it. :)

**Comment by josevalim** [comment]
@JustMikey which reference did you use to pick those values? Rails escapes many other characters: https://github.com/rails/rails/blob/master/actionview/lib/action_view/helpers/javascript_helper.rb#L6-L17

**Comment by JustMikey** [comment]
@josevalim ok :) Well, I played with it on practice and kind of added the bare minimum so it's probably up for improving.

**Pull request merged**